index.html

```diff
@@ -102,29 +109,79 @@ <h2>Items in cart</h2>
       function initializeCart() {
         const itemsInCart = []
 
+        const cartListElement = document.getElementById('cart-list')
+
+        function addToCart(array, item) {
+          array.push({
+          product: item.product,
+          price: item.price,
+          count: 1,
+          })
+        }
+
         Array.from(document.querySelectorAll('.add-to-cart')).forEach(
           (button) => {
             button.addEventListener('click', () => {
               console.log('clicked add to cart')
+              
+              const typesOfItemsInCart = itemsInCart.length
 
-              itemsInCart.push({
+              const latestCartItem = {
                 product: button.dataset.product,
                 price: Number(button.dataset.price),
-              })
-
-              const itemsInCartCount = itemsInCart.length
+              }
+
+              // This will keep track if the prodcut has been added to itemsInCart
+              let isInCart = false
+
+              // Check if the cart is empty
+              if (itemsInCart.length === 0) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              } else {
+                // Cart isn't empty, so is the same type of product already exist?
+                for (let i=0; i < typesOfItemsInCart; i++) {
+                  if (itemsInCart[i].product === latestCartItem.product) {
+                    itemsInCart[i].count++
```

Comment:
**Performance improvement suggestion** pt 1: Store the count in the `latestCartItem` object so that you don't need to loop through cart items again later on in order to get the count for the clicked product.

```suggestion
                    itemsInCart[i].count++
                    latestCartItem.count = itemsInCart[i].count
```